File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    static String[] alphabets = {"A","E","I","O","U"};
    static int ans = 0;
    static int cnt = 0;
    
    public int solution(String word) {
        int answer = 0;
        
        dfs("", word);
        
        return ans;
    }
    
    static void dfs(String str,String des){
        if(ans!=0) return;
        if(str.equals(des)){
            ans = cnt;
            return;
        }
        if(str.length() == 5) return;
        
        for(int i=0; i<5; i++){
            cnt++;
            dfs(str+alphabets[i], des);
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.kt
import java.util.*

val alphabets = listOf("A","E","I","O","U")
var ans = 0
var cnt = 0

class Solution {
    fun solution(word: String): Int {
        
        dfs(word, "", 5)
        
        return ans
    }
    
    fun dfs(target:String, str:String, size:Int){
        if(str==target){
            ans = cnt
            return
        }
        if(str.length == size) return
        
        for(i in 0 until 5){
            cnt++
            dfs(target, str+alphabets[i],size)
        }
    }
}